torch/fx/proxy.py

```diff
@@ -289,6 +289,17 @@ def create_arg(self, a: Any) -> Argument:
 
         Can be override to support more trace-specific types.
         """
+        # IMPORTANT: Are you here because you are trying to proxy a new type into
+        # the graph? Please Please Please contact someone on the PyTorch Compiler team;
+        # the considerations are subtle.
+        #
+        # 1) When you add a new type, all of the downstream consumers and pass writers
+        # need to handle the new type. torch.fx is intended to be easy to write
+        # passes for, so we will push back against new types.
+        # 2) In torch.compile's IR, there are only specific operations that go
+        # into the graph. In particular, Tensor operations should go into the graph,
+        # but non-Tensor operations shouldn't. What that means is that constructors
+        # for new types *SHOULD NOT* become nodes in the FX graph.
```

Comment:
But if non-Tensor operation (e.g., dataclass construction) is inlined into arguments of other nodes, wouldn't the IR consumer need to handle this extra case as well (in any logic that deals with node arguments).

Reply:
yes. Adding any type complicates the IR.

It's just that Dynamo (and the rest of the PT2 stack) use the (not very well enforced) invariant that nodes in the graph should only be tensor compute (with some exceptions)